### PR TITLE
support encryption for data store in disk

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
Signed-off-by: zhangjinpeng1987 <zhangjinpeng@pingcap.com>

## What have you changed? (mandatory)

- Encrypt data store in disk, including RocksDB's  WAL and SST files.

## What are the type of the changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:
- New feature

## How has this PR been tested? (mandatory)

- [x] performance test
- [x] correctness test

## Does this PR affect documentation (docs) update? (mandatory)

- [ ] security manual guide

## Does this PR affect tidb-ansible update? (mandatory)

No

Require https://github.com/pingcap/rust-rocksdb/pull/247 merged.

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
This PR is ready for review now.

**Comment by zhangjinpeng1987** [comment]
@huachaohuang @hicqu PTAL

File: src/raftstore/store/snap.rs [hunk 1]
```diff
@@ -712,16 +721,22 @@ impl Snap {
...
+                            let e = box_err!("cf_file is none for cf {}", cf);
+                            return Err(RaftStoreError::Snapshot(e));
+                        }
+                    };
+                    build_plain_cf_file(file, snap, cf, &begin_key, &end_key)?
                 };
-                build_plain_cf_file(file, snap, cf, &begin_key, &end_key)?
+                // Also update kv count for plain file.
+                let cf_file = &mut self.cf_files[self.cf_index];
+                cf_file.kv_count = kv_count as u64;
```
Comment (huachaohuang) [inline src/raftstore/store/snap.rs]: How about update it inside `build_plain_cf_file`? Seems more straightforward.

**Comment by zhangjinpeng1987** [comment]
@huachaohuang PTAL again
@hicqu PTAL

File: src/raftstore/store/snap.rs (hunk 1 above)
Comment (zhangjinpeng1987) [reply src/raftstore/store/snap.rs]: Done

File: src/raftstore/store/snap.rs [hunk 2]
```diff
@@ -650,7 +655,11 @@ impl Snap {
                 }
             }
             let size = get_file_size(&cf_file.tmp_path)?;
-            if size > 0 {
+            // The size of a sst file is larger than 0 doesn't mean it contains some key value pairs.
+            // For example, if we provide a encrypted env to the SstFileWriter, RocksDB will append
```
Comment (Connor1996) [inline src/raftstore/store/snap.rs]: Does it matter when rolling update new TiKV sends encrypted snapshots to TiKV without encryption enabled?

Comment (zhangjinpeng1987) [reply src/raftstore/store/snap.rs]: This is forbidden. This feature is design for new cluster, and doesn't support turn on/off for existing cluster.

**Pull request merged**